lua/telescope/_extensions/env.lua

```diff
@@ -1,82 +1,118 @@
-        end
-        local desc_width = math.floor(cols * 0.05)
-        local command_width = 22
+local function appendEnvironmentValue(prompt_bufnr)
+	local selection = action_state.get_selected_entry(prompt_bufnr)
+	actions.close(prompt_bufnr)
+	if selection.value == "" then
+		return
+	end
+	if vim.api.nvim_buf_get_option(vim.api.nvim_get_current_buf(), "modifiable") then
+		vim.api.nvim_put({ selection.value[2] }, "b", true, true)
+	end
+end
+
+local function appendEnvironmentName(prompt_bufnr)
+	local selection = action_state.get_selected_entry(prompt_bufnr)
+	actions.close(prompt_bufnr)
+	if selection.value == "" then
+		return
+	end
+	if vim.api.nvim_buf_get_option(vim.api.nvim_get_current_buf(), "modifiable") then
+		vim.api.nvim_put({ selection.value[1] }, "b", true, true)
+	end
+end
+
+local function editEnvironmentValue(prompt_bufnr)
+	local selection = action_state.get_selected_entry(prompt_bufnr)
+	actions.close(prompt_bufnr)
+	local value = vim.fn.input("[ENV] Enter new value: ", selection.value[2])
+	if value == "" then
+		return
+	end
+	print("Setting env %s=%s", selection.value[1], value)
+	-- Only works for the current session
+	vim.fn.setenv(selection.value[1], value)
+end
 
-        -- NOTE: the width calculating logic is not exact, but approx enough
-        local displayer = entry_display.create {
-          separator = " ▏",
-          items = {
-            { width = command_width },
-            { width = tel_win_width - desc_width - command_width },
-            { remaining = true },
-          },
-        }
+local function showEnvironmentVariables(opts)
+	opts = opts or {}
+	pickers.new(opts, {
+		prompt_title = "Environment Variables",
+		finder = finders.new_table({
+			results = prepareEnvironmentVariables(),
+			entry_maker = function(entry)
+				local cols = vim.o.columns
+				local width = conf.width
+					or conf.layout_config.width
+					or conf.layout_config[conf.layout_strategy].width
+					or cols
+				local tel_win_width
+				if width > 1 then
+					tel_win_width = width
+				else
+					tel_win_width = math.floor(cols * width)
+				end
+				local desc_width = math.floor(cols * 0.05)
+				local command_width = 22
 
-        local function make_display(ent)
-          return displayer
-          {
-            { entry[1] },
-            { entry[2] },
-          }
-        end
+				-- NOTE: the width calculating logic is not exact, but approx enough
+				local displayer = entry_display.create({
+					separator = " ▏",
+					items = {
+						{ width = command_width },
+						{ width = tel_win_width - desc_width - command_width },
+						{ remaining = true },
+					},
+				})
 
-        return
-        {
-          value = entry,
-          display =  make_display,
-          ordinal = string.format('%s %s', entry[1], entry[2])
-        }
-      end,
+				local function make_display(ent)
+					-- Concatinate multiline env values
+					local concat_value = entry[2]:gsub("\r?\n", " ")
```

Comment:
> fixes weird behavior (bug?) with multiline values

This line fixed this issue?

I don't see any difference in behaviour between my implementation and yours. Can you please elaborate more on this?

Reply:
Sure.
I use fish(not sure if the same bug exists on bash) and when the multiline environmental variable is defined, i.e.:
```
FZF_CTRL_T_COMMAND = fd --type file \
  --exclude ".git" \
  --exclude "venv*" \
  --exclude "*node_modules*" \
  --exclude "*ctags*" \
  --exclude "Library" \
  --exclude "Music" 
```

`C-n` and `C-p` shortcuts sometimes are not working, not all entries are displayed. See this demo:

https://user-images.githubusercontent.com/28604639/149325871-391b4c17-2d2b-44a8-a630-ccaff22b278e.mov

With concatenated entries problem is gone

Reply:
Thank you.